File: src/integrations/api/vendasProspeccao.ts
/**
 * Client de Vendas — Prospecção (Toriq Vendas Fase 1).
 *
 * Cobre configuração do Apify (token + actors) e disparo/acompanhamento de jobs
 * de scraping, batendo com o backend FastAPI (`/vendas`, snake_case). Usa o
 * client interno `@/integrations/api/client`, que envia o cookie httpOnly de
 * auth e faz refresh transparente em 401.
 *
 * Complementa `@/integrations/api/vendas` (Fase 0 — leads/tags/segmentação).
 */

import { api } from "@/integrations/api/client";

// ---------------------------------------------------------------------------
// Tipos (snake_case, espelhando o backend)
// ---------------------------------------------------------------------------

/** Configuração pública (o token nunca é devolvido em claro). */
export interface ProspeccaoConfig {
  apify_token_set: boolean;
  apify_token_masked: string | null;
  actors: Record<string, string> | null;
}

/** Payload de atualização da config. */
export interface ProspeccaoConfigUpdate {
  apify_token?: string;
  actors?: Record<string, string>;
  clear_apify_token?: boolean;
}

/** Job de scraping. */
export interface ProspeccaoJob {
  id: string;
  empresa_id: string;
  plataforma: string;
  parametros: Record<string, unknown> | null;
  tag_id: string | null;
  apify_run_id: string | null;
  apify_dataset_id: string | null;
  status: string;
  total_captados: number;
  total_importados: number;
  total_duplicados: number;
  custo: number | null;
  erro: string | null;
  created_at: string | null;
  updated_at: string | null;
  finished_at: string | null;
}

export interface ProspeccaoStatus {
  job_id: string;
  status: string;
  total_captados: number;
}

export interface ProspeccaoResults {
  inseridos: number;
  duplicados: number;
  total: number;
}

// ---------------------------------------------------------------------------
// API
// ---------------------------------------------------------------------------

export const prospeccaoApi = {
  // Configuração ---------------------------------------------------------
  getConfig: () => api.get<ProspeccaoConfig>("/vendas/config"),

  saveConfig: (data: ProspeccaoConfigUpdate) =>
    api.put<ProspeccaoConfig>("/vendas/config", data),

  // Scraping -------------------------------------------------------------
  startScraping: (
    plataforma: string,
    parametros: Record<string, unknown>,
    tag_nome?: string,
  ) =>
    api.post<ProspeccaoJob>("/vendas/scraping/start", {
      plataforma,
      parametros,
      tag_nome,
    }),

  checkStatus: (job_id: string) =>
    api.post<ProspeccaoStatus>("/vendas/scraping/status", { job_id }),

  fetchResults: (job_id: string, force = false) =>
    api.post<ProspeccaoResults>("/vendas/scraping/results", { job_id, force }),

  // Jobs -----------------------------------------------------------------
  listJobs: (limit = 20) =>
    api.get<ProspeccaoJob[]>(`/vendas/jobs?limit=${limit}`),

  cancelJob: (job_id: string) =>
    api.post<ProspeccaoJob>(`/vendas/jobs/${job_id}/cancel`),
};

File: apps/api/app/schemas/vendas_prospeccao.py
"""Schemas Pydantic para o módulo Toriq Vendas — FASE 1 (Prospecção via Apify).

snake_case batendo com os models/colunas do backend. Escopo: configuração do token
Apify + disparo de actors de scraping + importação dos resultados em vendas_leads.
Tenant SEMPRE por user.empresa_id (carimbado no router).
"""
import datetime
import uuid
from typing import Optional

from pydantic import BaseModel, ConfigDict


# ── Config ──────────────────────────────────────────────────────────────────

class ConfigUpdate(BaseModel):
    apify_token: Optional[str] = None
    actors: Optional[dict] = None
    cache_dias: Optional[int] = None
    clear_apify_token: Optional[bool] = None


class ConfigPublic(BaseModel):
    apify_token_set: bool
    apify_token_masked: Optional[str] = None
    actors: Optional[dict] = None
    cache_dias: Optional[int] = None
    model_config = ConfigDict(from_attributes=True)


# ── Scraping ──────────────────────────────────────────────────────────────────

class ScrapingStartIn(BaseModel):
    plataforma: str
    parametros: dict
    tag_nome: Optional[str] = None


class ScrapingStatusIn(BaseModel):
    job_id: uuid.UUID


class ScrapingStatusOut(BaseModel):
    job_id: uuid.UUID
    status: str
    total_captados: int


class ScrapingResultsIn(BaseModel):
    job_id: uuid.UUID
    force: bool = False


class ScrapingResultsOut(BaseModel):
    inseridos: int
    duplicados: int
    total: int


# ── Jobs ──────────────────────────────────────────────────────────────────────

class JobOut(BaseModel):
    id: uuid.UUID
    empresa_id: uuid.UUID
    plataforma: str
    parametros: Optional[dict]
    tag_id: Optional[uuid.UUID]
    apify_run_id: Optional[str]
    apify_dataset_id: Optional[str]
    status: str
    total_captados: int
    total_importados: int
    total_duplicados: int
    custo: Optional[float]
    from_cache: Optional[bool] = False
    erro: Optional[str]
    created_at: Optional[datetime.datetime]
    updated_at: Optional[datetime.datetime]
    finished_at: Optional[datetime.datetime]
    model_config = ConfigDict(from_attributes=True)
